compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5615,10 +5615,10 @@ private boolean populateFields(IntersectionContext intersectionContext, BRecordT
                 invokableSymbol.flags = tsymbol.flags;
             } else {
                 recordFieldSymbol = new BVarSymbol(intersectionFlags, name, env.enclPkg.packageID,
-                                                   intersectionFieldType, newTypeSymbol, lhsRecordField.pos, SOURCE);
+                        intersectionFieldType, newTypeSymbol, lhsRecordField.symbol.pos, SOURCE);
             }
 
-            newTypeFields.put(key, new BField(name, recordFieldSymbol.pos, recordFieldSymbol));
+            newTypeFields.put(key, new BField(name, null, recordFieldSymbol));
```

Comment:
I think you got what I said wrong. What I meant was, whether we can remove the `pos` field from the `BField`. If we can't, then let's just use `recordFieldSymbol.pos` for `pos` here.

Reply:
Currently, there are multiple usages that use the `pos` field of the `BField`. We may require a separate issue to evaluate and refactor this change if applicable.